nipoppy_cli/nipoppy/cli/parser.py

```diff
@@ -141,15 +147,19 @@ def _verbosity_to_log_level(verbosity: str):
         try:
             return VERBOSITY_TO_LOG_LEVEL_MAP[verbosity]
         except KeyError:
-            parser.error(
-                f"Invalid verbosity level: {verbosity}."
-                f" Valid levels are {list(VERBOSITY_TO_LOG_LEVEL_MAP.keys())}."
+            raise ArgumentError(
+                None,
+                (
+                    f"Invalid verbosity level: {verbosity}. "
+                    f"Valid levels are: {', '.join(VERBOSITY_TO_LOG_LEVEL_MAP.keys())}."
+                ),
             )
 
     parser.add_argument(
         "--verbosity",
         type=_verbosity_to_log_level,
         default=DEFAULT_VERBOSITY,
```

Comment:
I was wondering why it was not originally chosen to do something like

```python
    update_parser.add_argument(
        "--verbosity",
        help="""
        Verbosity level.
        """,
        required=False,
        choices=list(VERBOSITY_TO_LOG_LEVEL_MAP.keys()),
        default=DEFAULT_VERBOSITY,
        type=int,
        nargs=1,
    )
```

this would take care of the validattion, no?

Reply:
I guess we could do it this way and have the mapping processing be done in `cli/run.py` instead